lib/kino_explorer/data_transform_cell.ex

```diff
@@ -426,14 +425,25 @@ defmodule KinoExplorer.DataTransformCell do
       |> Enum.map(&Map.new(&1, fn {k, v} -> convert_field(k, v) end))
       |> Enum.chunk_by(& &1.operation_type)
       |> Enum.map(&(to_quoted(&1) |> Map.merge(%{module: attrs.data_frame_alias})))
+      |> Enum.filter(& &1.args)
 
-    has_pivot_wider = Enum.any?(nodes, &(&1.name == :pivot_wider && &1.args))
+    pivot_wider = Enum.find_index(nodes, &(&1.name == :pivot_wider))
+
+    group =
+      nodes
+      |> Enum.with_index()
+      |> Enum.map(fn {v, i} -> if v.name == :group_by, do: i end)
+      |> Enum.reject(&(!&1 or &1 >= length(nodes) - 1))
+      |> Enum.map(&if Map.get(Enum.at(nodes, &1 + 1), :name) != :summarise, do: &1)
+      |> List.first()
```

Comment:
I wonder if this would be simpler to implement with a recursive function. You call it as `collect_index(nodes, 0)` and:

```
defp collect_index([%{name: :group_by}, %{name: :summarise} | rest], idx),
  do: collect_index(rest, idx + 2)

defp collect_index([%{name: :group_by} | _], idx) when name in [:group_by, :pivot_wider],
  do: idx

defp collect_index([_ | rest], idx),
  do: collect_index(rest, idx + 1)

defp collect_index([], idx),
  do: idx
```

The above should handle both group_by and pivot_wider, in a single pass, and I find it clearer too. WDYT?